### PR TITLE
Simplify check hasDependencies and add unit tests

### PR DESCRIPTION
**Pull request by Connormiha**
**What kind of change does this PR introduce?**
Added unit tests + minor refactoring

**Did you add tests for your changes?**
Yes

**Summary**
Added different test cases for `hasDependencies`.
Simplify mehod.

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

**Comment by ooflorent** [comment]
@Connormiha you could ignore the failure.

**Comment by sokra** [comment]
Unit tests do not contribute to test coverage. To test this add an integration test which covers this scenario.

**Comment by Connormiha** [comment]
@sokra 
It is very strange that the code coverage has decreased. I just cut a few strings (I did't add new features).

File: lib/DependenciesBlockVariable.js [hunk 1]
```diff
@@ -40,11 +40,10 @@ class DependenciesBlockVariable {
 
 	hasDependencies(filter) {
 		if(filter) {
-			if(this.dependencies.some(filter)) return true;
+			return this.dependencies.some(filter);
 		} else {
-			if(this.dependencies.length > 0) return true;
+			return this.dependencies.length > 0;
```
Comment (krzysiek1507) [inline lib/DependenciesBlockVariable.js]: You can omit `else` block and just type this statement on the end of the function

**Comment by Connormiha** [comment]
@ooflorent should i do something else to avoid CI warnings at this PR( i think these were falsely warnings)?

**Comment by ooflorent** [comment]
@Connormiha the warning is fine. I've removed it.

**Comment by sokra** [comment]
Thanks

**Pull request merged**